overlord/snapstate/snapmgr.go

```diff
@@ -249,10 +251,11 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 			return err
 		}
 	}
+	si.Revision = ss.Revision
```

Comment:
Yeah, tests missing, I think the direction is relatively clear. I add proper tests to avoid silly issues like t his.